engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java

```diff
@@ -37,18 +39,21 @@ public final class DbJobState implements JobState, MutableJobState {
   private final JobRecordValue jobRecordToWrite = new JobRecordValue();
 
   private final DbLong jobKey;
-  private final DbString tenantIdKey;
   private final DbForeignKey<DbLong> fkJob;
   private final ColumnFamily<DbLong, JobRecordValue> jobsColumnFamily;
 
   // key => job state
   private final JobStateValue jobState = new JobStateValue();
   private final ColumnFamily<DbForeignKey<DbLong>, JobStateValue> statesJobColumnFamily;
 
-  // type => [key]
+  // [[type, key], tenant_id] => nil
   private final DbString jobTypeKey;
+  private final DbString tenantIdKey;
   private final DbCompositeKey<DbString, DbForeignKey<DbLong>> typeJobKey;
-  private final ColumnFamily<DbCompositeKey<DbString, DbForeignKey<DbLong>>, DbNil>
+  private final DbCompositeKey<DbCompositeKey<DbString, DbForeignKey<DbLong>>, DbString>
```

Comment:
❌ We can use the `DbTenantAwareKey` with `PrefixType.SUFFIX` here:

`DbTenantAwareKey<DbCompositeKey<DbString, DbForeignKey<DbLong>>`

Reply:
@remcowesterhoud with @korthout we considered using the `DbTenantAwareKey` here, but it turned out that it would be more difficult to apply the workaround suggested [here](https://camunda.slack.com/archives/C05Q0EXE9V4/p1695223551936369). This way, it's clearer that we've implemented a workaround instead of a proper tenant-aware structure.

Reply:
I don't see why it would be more difficult that way. But as I wasn't there and you discussed it with @korthout I'm sure there was a good reason for it. We can always refactor it later.

Reply:
It's a good question @remcowesterhoud.

The workaround would seep into the `DbTenantAwareKey`. That means the `DbTenantAwareKey` would be more flexible which is great, but it would also lose some of the guarantees it offers now (we know that a tenantId exists there). It would also either make `DbTenantAwareKey` more complex, or have different guarantees for `PREFIX/SUFFIX`. At this time, I still believe it's best to keep this a separate situation. 

Of course the column family is documented so that it's clear that a tenantId is involved 👍

Reply:
Thanks for clarifying @korthout. So it's not about difficulty per se, it's more about keeping `DbTenantAwareKey` code clean 👍

Reply:
Yes, and the expectations when using it in other places. See the difficulties with activatableColumnFamily here: https://github.com/camunda/zeebe/pull/14407#discussion_r1338298045